packages/gasket-request/lib/wrappers.js

```diff
@@ -0,0 +1,26 @@
+import { GasketRequest } from './request.js';
+
+/** @type {import('./internal.d.ts').withGasketRequestCache} */
+export function withGasketRequestCache(actionFn) {
+  const cache = new WeakMap();
+
+  return async (gasket, req, ...args) => {
+    const normalizedReq = await GasketRequest.make(req);
+
+    if (cache.has(normalizedReq)) {
+      return cache.get(normalizedReq);
+    }
+
+    const result = await actionFn(gasket, normalizedReq, ...args);
+    cache.set(normalizedReq, result);
+    return result;
```

Comment:
Similar feedback here; you're not going to be deduplicating parallel requests, only serial ones, unless you also cache the Promise. This should fix that issue.

```suggestion
    if (cache.has(normalizedReq)) {
      return cache.get(normalizedReq);
    }

    const promise = actionFn(gasket, normalizedReq, ...args);
    cache.set(normalizedReq, promise);
    
    try {
      const result = await promise;
      cache.set(normalizedReq, result);
    } catch (err) {
      cache.delete(normalizedReq);
      throw err;
    }

    return result;
```